octodns/record/__init__.py

```diff
@@ -1091,7 +1091,10 @@ def validate(cls, data, _type):
             exchange = None
             try:
                 exchange = value.get('exchange', None) or value['value']
```

Comment:
Do we also want to do type checking or casting to str here? If there's a non-str, the else condition will throw an exception.

Reply:
Good question. I just copy-n-pasted the existing `_TargetValue` code block which does the `str`. 

Coming from Yaml I think the only other types that could happen are: int, float, list, or dict. All of those would stringify cleanly and fail the `FQDN` check so I think this might be OK as-is, though maybe not an ideal error message. 

You see any other cases?

There's probably a TON of places we could do better error messages for unexpected types. Not opposed to it here if the existing invalid FQDN message isn't sufficient.

Reply:
Good point, I think the FQDN error will suffice.